### ISSUE TITLE
gp_sampler and opt_gp_sampler

### ISSUE THREAD

**Issue by oxinabox** [COBREXA/COBREXA.jl#61]
## Target functionality

Currently Cobrexa implements 1 method for [flux sampling](https://cobrexa.github.io/COBREXA.jl/stable/examples/06a-sampling/): 
ACHR.
This is set by passing the `method` argument to [`flux_sample`](https://cobrexa.github.io/COBREXA.jl/stable/reference/frontend/#COBREXA.flux_sample-Tuple{AbstractFBCModels.AbstractFBCModel}) but there is no other options provid)ed in the package other than [`sample_chain_achr`](https://cobrexa.github.io/COBREXA.jl/stable/reference/analysis/#COBREXA.sample_chain_achr-Union{Tuple{SM},%20Tuple{M},%20Tuple{V},%20Tuple{F}}%20where%20{F%3C:Real,%20V%3C:AbstractVector{F},%20M%3C:AbstractMatrix{F},%20SM%3C:AbstractMatrix{F}}).

There are two methods which sucessively build on ACHR:

- [ ] `gp_sampler`
- [ ] `opt_gp_sampler` (https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0086587

## Desired output

- `flux_sample(model,  method=gp_sampler)`
- `flux_sample(model,  method=opt_gp_sampler)`

**Comment by exaexa** [comment]
@stelmo is it just me or (opt)gpSampler improvements are so negligible that our ACHR impl already does them? In the paper I can't detect anything that we don't do already.

**Comment by oxinabox** [comment]
Once that update/port is done, updating the docs on that page to reference that package.
It would be nice to have docs on what the API for the `method` argument is meant to tag.
Though not pressing

**Comment by stelmo** [comment]
> @stelmo is it just me or (opt)gpSampler improvements are so negligible that our ACHR impl already does them? In the paper I can't detect anything that we don't do already.

Yep, I am not sure there is much benefit in extending sampling functionality beyond what we have right now.
